src/server.rs

```diff
@@ -732,3 +770,343 @@ async fn handle_hexists(
 
     Ok(())
 }
+
+pub fn compress(
```

Comment:
should be traits, and impl for each implementation,
|_ compression
    |_ gzip.rs
    |_ zstd.rs